File: 1.12/solution.exs
defmodule Binom do
  def calc(row) do
    calc_iter(row, 1)
  end

  def calc_iter(row, cnt) when cnt > row, do: IO.puts "#{calc_elem(row, cnt)} "
  def calc_iter(row, cnt) do
    IO.write "#{calc_elem(row, cnt)} "
    calc_iter(row, cnt + 1)
  end

  def calc_elem(row, element) when element < 1 or element > row + 1, do: 0

  def calc_elem(1, _) do
    1
  end

  def calc_elem(row, element) do
    calc_elem(row - 1, element - 1) + calc_elem(row - 1, element)
  end
end

Enum.each(1..10, fn(degree) ->
  IO.puts "Degree: #{degree}"
  Binom.calc(degree)
  IO.puts "=================="
end)

File: 1.12/solution.rs
fn main() {
    for n in 1..=10 {
        println!("Degree: {}", n);
        calc(n);
        println!("================");
    }
}

fn calc(row: i32) {
    calc_iter(row, 1);
}

fn calc_iter(row: i32, cnt: i32) {
    if cnt > row {
        println!("{}", calc_elem(row, cnt));
    } else {
        print!("{} ", calc_elem(row, cnt));
        calc_iter(row, cnt + 1);
    }
}

fn calc_elem(row: i32, element: i32) -> i32 {
    if element < 1 || element > row + 1 {
        return 0;
    } else if row == 1 {
        return 1;
    } else {
        return calc_elem(row - 1, element - 1) + calc_elem(row - 1, element);
    }
}